[RUM-8706] [V3] Expose `setAccountInfo` APIs

cdn34dd opened the pull request:
### What does this PR do?

Exposes setAccountInfo APIs to JS layer.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [ ] If this PR is auto-generated, please make sure also to manually update the code related to the change

cdn34dd committed:
Expose `setAccountInfo` API to JS layer

add account info to `applyEventMapper`

sbarrio commented on packages/core/src/sdk/AccountInfoSingleton/AccountInfoSingleton.ts:
```diff
@@ -0,0 +1,46 @@
+        };
+    };
+
+    getAccountInfo = (): AccountInfo | undefined => {
```
Should we add this accountInfo to `applyEventMapper` like [we do with UserInfo and Attributes](https://github.com/DataDog/dd-sdk-reactnative/blob/bcfa888bad40812cf57ce56ad1ecd89896a62925/packages/core/src/sdk/EventMappers/EventMapper.ts#L68-L73)?

If not, we may consider removing this Singleton altogether and just rely on the native layer to store and propagate this info forward.

cdn34dd replied:
Yeah I believe we should, I forgot to add it there.

cdn34dd committed:
Add `userId` and `accountId`to baggage headers

Preserve user baggage header when setting session ID

Enforced W3 specification for 'baggage' header

Only inject Session ID header if propagator=Datadog|W3C

Additional test for baggage header and minor warn message improvement

cdn34dd committed:
Cache `userId` and `accountId` when first set

sbarrio reviewed (approved):
🚀
